packages/mobile/src/redux/store.test.ts

```diff
@@ -28,11 +44,356 @@ describe('persistConfig', () => {
   })
 
   it('migrate calls the expected methods', async () => {
-    mockedCreateMigrate.mockReturnValue(() => Promise.resolve({ migrated: true }) as any)
-    mockedResetStateOnInvalidStoredAccount.mockImplementation((state) => Promise.resolve(state))
+    createMigrate.mockReturnValue(() => Promise.resolve({ migrated: true }) as any)
     const state = await _persistConfig.migrate!({} as any, -1)
     expect(state).toEqual({ migrated: true })
     expect(createMigrate).toHaveBeenCalledTimes(1)
     expect(resetStateOnInvalidStoredAccount).toHaveBeenCalledTimes(1)
   })
 })
+
+describe('store state', () => {
+  // This test ensures the vNeg1Schema can be successfully migrated to the latest version
+  // and validates against the latest RootState schema
+  // See https://github.com/valora-inc/wallet/tree/main/packages/mobile#redux-state-migration
+  it('validates against the RootState schema after rehydration', async () => {
+    const { store, persistor } = setupStore(undefined, {
+      ..._persistConfig,
+      // @ts-ignore
+      getStoredState: async () => {
+        // Remove undefined values from the test state
+        return JSON.parse(JSON.stringify(vNeg1Schema))
+      },
+    })
+
+    // Wait for the persistor to have rehydrated the state
+    await new Promise((resolve) => {
+      persistor.subscribe(() => {
+        const { bootstrapped } = persistor.getState()
+        if (bootstrapped) {
+          resolve(true)
+        }
+      })
+    })
+
+    const data = store.getState()
+
+    const ajv = new Ajv({ allErrors: true })
+    const schema = require('test/RootStateSchema.json')
+    const validate = ajv.compile(schema)
+    const isValid = validate(data)
+
+    // console.log('Validation errors:', validate.errors?.length, validate.errors)
+    expect(validate.errors).toBeNull()
+    expect(isValid).toBe(true)
+
+    // Make sure the state looks the way we expect it to.
+    // This should help catch unintentional changes to the state, caused by incorrect migrations
+    expect(data).toMatchInlineSnapshot(`
```

Comment:
if i understand correctly, this snapshot also needs to be updated every time there is a redux migration? just wondering about the value of this, as we generally don't pay that much attention to snapshot value changes 🤔 for me the important part of validating the root state schema is already done in the previous 2 assertions, i suppose this assertion gives us confidence in the initial values of each property...but i'm wondering if the extra effort is worth it for this. what do you think?

Reply:
Without the snapshot there's a chance we could reset some values, for instance loosing the account or other important user settings.
The state would still be of the right shape but without the expected persisted values. I thought having the snapshot was one more way to ensure this doesn't happen.